src/main/java/uk/gov/pay/adminusers/queue/event/EventMessageHandler.java

```diff
@@ -1,42 +1,122 @@
+import uk.gov.pay.adminusers.model.Service;
+import uk.gov.pay.adminusers.persistence.dao.UserDao;
+import uk.gov.pay.adminusers.persistence.entity.RoleEntity;
+import uk.gov.pay.adminusers.persistence.entity.UserEntity;
+import uk.gov.pay.adminusers.queue.model.Event;
 import uk.gov.pay.adminusers.queue.model.EventMessage;
+import uk.gov.pay.adminusers.queue.model.EventType;
+import uk.gov.pay.adminusers.queue.model.event.DisputeCreatedDetails;
+import uk.gov.pay.adminusers.service.NotificationService;
+import uk.gov.pay.adminusers.service.ServiceFinder;
 import uk.gov.service.payments.commons.queue.exception.QueueException;
 
 import javax.inject.Inject;
+import java.time.format.DateTimeFormatter;
+import java.util.Collections;
 import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
+import static java.lang.String.format;
 import static net.logstash.logback.argument.StructuredArguments.kv;
+import static uk.gov.pay.adminusers.utils.currency.ConvertToCurrency.convertPenceToPounds;
+import static uk.gov.pay.adminusers.utils.date.DisputeEvidenceDueByDateUtil.getPayDueByDateForEpoch;
+import static uk.gov.pay.adminusers.utils.date.DisputeEvidenceDueByDateUtil.getZDTForEpoch;
 
 public class EventMessageHandler {
-    
+
     private final Logger logger = LoggerFactory.getLogger(getClass());
     private final EventSubscriberQueue eventSubscriberQueue;
+    private final LedgerService ledgerService;
+    private final NotificationService notificationService;
+    private final ServiceFinder serviceFinder;
+    private final UserDao userDao;
+    private final ObjectMapper objectMapper;
+    private final DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("dd, LLLL yyyy"); // 09, March 2022
 
     @Inject
-    public EventMessageHandler(EventSubscriberQueue eventSubscriberQueue) {
+    public EventMessageHandler(EventSubscriberQueue eventSubscriberQueue, LedgerService ledgerService, NotificationService notificationService, ServiceFinder serviceFinder, UserDao userDao, ObjectMapper objectMapper) {
         this.eventSubscriberQueue = eventSubscriberQueue;
+        this.ledgerService = ledgerService;
+        this.notificationService = notificationService;
+        this.serviceFinder = serviceFinder;
+        this.userDao = userDao;
+        this.objectMapper = objectMapper;
     }
-    
+
     public void processMessages() throws QueueException {
         List<EventMessage> eventMessages = eventSubscriberQueue.retrieveEvents();
         for (EventMessage message : eventMessages) {
             try {
                 logger.info("Retrieved event queue message with id {} for resource external id {}",
                         message.getQueueMessage().getMessageId(), message.getEvent().getResourceExternalId());
-                
+                if (message.getEvent().getEventType().equalsIgnoreCase(EventType.DISPUTE.name())) {
```

Comment:
Won't the event type be "DISPUTE_CREATED"?

Reply:
Good spot

Reply:
Fixed